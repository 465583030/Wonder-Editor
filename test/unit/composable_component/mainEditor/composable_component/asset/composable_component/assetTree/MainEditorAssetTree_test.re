open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe(
    "MainEditorAssetTree",
    () => {
      let sandbox = getSandboxDefaultVal();
      beforeEach(
        () => {
          sandbox := createSandbox();
          MainEditorSceneTool.initStateAndGl(~sandbox, ())
        }
      );
      afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
      describe(
        "test show component",
        () => {
          test(
            "test simple assetTree which haven't children case",
            () => {
              MainEditorSceneTool.createDefaultScene(
                sandbox,
                MainEditorAssetTool.initAssetTree(MainEditorAssetTool.buildSimpleAssetTree)
              );
              BuildComponentTool.buildAssetTreeComponent() |> ReactTestTool.createSnapshotAndMatch
            }
          );
          describe(
            "test has children case",
            () => {
              describe(
                "have first layer children",
                () => {
                  beforeEach(
                    () =>
                      MainEditorSceneTool.createDefaultScene(
                        sandbox,
                        MainEditorAssetTool.initAssetTree(
                          MainEditorAssetTool.buildTwoLayerAssetTree
                        )
                      )
                  );
                  test(
                    "no drag",
                    () =>
                      BuildComponentTool.buildAssetTreeComponent()
                      |> ReactTestTool.createSnapshotAndMatch
                  );
                  test(
                    "drag treeNode into borther treeNode",
                    () => {
                      let component = BuildComponentTool.buildAssetTreeComponent();
                      BaseEventTool.triggerComponentEvent(
                        component,
                        AssetTreeEventTool.triggerFirstLayerDragStartEvent(2)
                      );
                      BaseEventTool.triggerComponentEvent(
                        component,
                        AssetTreeEventTool.triggerFirstLayerDragEnterEvent(1)
                      );
                      BaseEventTool.triggerComponentEvent(
                        component,
                        AssetTreeEventTool.triggerFirstLayerDropEvent(1)
                      );
                      BuildComponentTool.buildAssetTreeComponent()
                      |> ReactTestTool.createSnapshotAndMatch
                    }
                  )
                }
              );
              describe(
                "have second layer children",
                () => {
                  beforeEach(
                    () =>
                      MainEditorSceneTool.createDefaultScene(
                        sandbox,
                        MainEditorAssetTool.initAssetTree(
                          MainEditorAssetTool.buildThreeLayerAssetTree
                        )
                      )
                  );
                  test(
                    "no drag",
                    () =>
                      BuildComponentTool.buildAssetTreeComponent()
                      |> ReactTestTool.createSnapshotAndMatch
                  );
                  test(
                    "drag second treeNode into root treeNode",
                    () => {
                      let component = BuildComponentTool.buildAssetTreeComponent();
                      BaseEventTool.triggerComponentEvent(
                        component,
                        AssetTreeEventTool.triggerSecondLayerDragStartEvent(1, 1)
                      );
                      BaseEventTool.triggerComponentEvent(
                        component,
                        AssetTreeEventTool.triggerRootDragEnterEvent
                      );
                      BaseEventTool.triggerComponentEvent(
                        component,
                        AssetTreeEventTool.triggerRootDropEvent
                      );
                      BuildComponentTool.buildAssetTreeComponent()
                      |> ReactTestTool.createSnapshotAndMatch
                    }
                  )
                }
              )
            }
          )
        }
      );
      describe(
        "deal with the specific case",
        () => {
          test(
            "if drag treeNode into itself, keep not change",
            () => {
              MainEditorSceneTool.createDefaultScene(
                sandbox,
                MainEditorAssetTool.initAssetTree(MainEditorAssetTool.buildTwoLayerAssetTree)
              );
              let component = BuildComponentTool.buildAssetTreeComponent();
              BaseEventTool.triggerComponentEvent(
                component,
                AssetTreeEventTool.triggerFirstLayerDragStartEvent(1)
              );
              BaseEventTool.triggerComponentEvent(
                component,
                AssetTreeEventTool.triggerFirstLayerDragEnterEvent(1)
              );
              BaseEventTool.triggerComponentEvent(
                component,
                AssetTreeEventTool.triggerFirstLayerDropEvent(1)
              );
              BuildComponentTool.buildAssetTreeComponent() |> ReactTestTool.createSnapshotAndMatch
            }
          );
          test(
            "if drag treeNode into it's chidlren, keep not change",
            () => {
              MainEditorSceneTool.createDefaultScene(
                sandbox,
                MainEditorAssetTool.initAssetTree(MainEditorAssetTool.buildThreeLayerAssetTree)
              );
              let component = BuildComponentTool.buildAssetTreeComponent();
              BaseEventTool.triggerComponentEvent(
                component,
                AssetTreeEventTool.triggerFirstLayerDragStartEvent(1)
              );
              BaseEventTool.triggerComponentEvent(
                component,
                AssetTreeEventTool.triggerSecondLayerDragEnterEvent(1, 1)
              );
              BaseEventTool.triggerComponentEvent(
                component,
                AssetTreeEventTool.triggerSecondLayerDropEvent(1, 1)
              );
              BuildComponentTool.buildAssetTreeComponent() |> ReactTestTool.createSnapshotAndMatch
            }
          )
        }
      )
    }
  );