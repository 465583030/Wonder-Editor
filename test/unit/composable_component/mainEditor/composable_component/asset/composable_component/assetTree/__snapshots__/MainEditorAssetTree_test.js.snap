// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetTree deal with the specific case if drag treeNode into it's chidlren, keep not change 1`] = `
<article
  className="tree-content"
>
  <ul
    className="wonder-tree-node"
  >
    <li
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <img
        src="./public/img/12.jpg"
      />
      asset
    </li>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder1
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "border": "1px solid red",
            }
          }
        >
          <img
            src="./public/img/12.jpg"
          />
          folder2
        </li>
      </ul>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetTree deal with the specific case if drag treeNode into itself, keep not change 1`] = `
<article
  className="tree-content"
>
  <ul
    className="wonder-tree-node"
  >
    <li
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <img
        src="./public/img/12.jpg"
      />
      asset
    </li>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder1
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder2
      </li>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetTree test show component test has children case have first layer children drag treeNode into borther treeNode 1`] = `
<article
  className="tree-content"
>
  <ul
    className="wonder-tree-node"
  >
    <li
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <img
        src="./public/img/12.jpg"
      />
      asset
    </li>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder1
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "border": "1px solid red",
            }
          }
        >
          <img
            src="./public/img/12.jpg"
          />
          folder2
        </li>
      </ul>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetTree test show component test has children case have first layer children no drag 1`] = `
<article
  className="tree-content"
>
  <ul
    className="wonder-tree-node"
  >
    <li
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <img
        src="./public/img/12.jpg"
      />
      asset
    </li>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder1
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder2
      </li>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetTree test show component test has children case have second layer children drag second treeNode into root treeNode 1`] = `
<article
  className="tree-content"
>
  <ul
    className="wonder-tree-node"
  >
    <li
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <img
        src="./public/img/12.jpg"
      />
      asset
    </li>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder1
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder2
      </li>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetTree test show component test has children case have second layer children no drag 1`] = `
<article
  className="tree-content"
>
  <ul
    className="wonder-tree-node"
  >
    <li
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <img
        src="./public/img/12.jpg"
      />
      asset
    </li>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "border": "1px solid red",
          }
        }
      >
        <img
          src="./public/img/12.jpg"
        />
        folder1
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "border": "1px solid red",
            }
          }
        >
          <img
            src="./public/img/12.jpg"
          />
          folder2
        </li>
      </ul>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetTree test show component test simple assetTree which haven't children case 1`] = `
<article
  className="tree-content"
>
  <ul
    className="wonder-tree-node"
  >
    <li
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <img
        src="./public/img/12.jpg"
      />
      asset
    </li>
  </ul>
</article>
`;
